src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs

```diff
@@ -127,5 +127,94 @@ public static void ValidatePropertyAndValue(ReadOnlySpan<char> propertyName, Rea
             if (propertyName.Length > JsonConstants.MaxCharacterTokenSize || value.Length > JsonConstants.MaxCharacterTokenSize)
                 ThrowHelper.ThrowArgumentException(propertyName, value);
         }
+
+        internal static void ValidateNumber(ReadOnlySpan<byte> utf8FormattedNumber)
```

Comment:
Is this check really necessary? It seems like redundant work to me. The reader already validated it.

Reply:
Yes, for two reasons.

1) The memory that the DOM is reading from is still user controlled, and not immutable.  So there's no guarantee that what was a number is still a number.  Nothing else from the DOM would be capable of producing a corrupt output (true/false/null all ignore the original input; strings and property names are quoted, objects and arrays are properly bounded).

2) The check should be there for public API, seemed like it was best to just write it in now, in case you wish to pursue it as public API later.